http/java/okhttp/lib/src/main/java/com/microsoft/kiota/http/middleware/TelemetryHandler.java

```diff
@@ -0,0 +1,62 @@
+package com.microsoft.kiota.http.middleware;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.microsoft.kiota.http.middleware.options.TelemetryHandlerOption;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * TelemetryHandler implementation using OkHttp3
+ */
+public class TelemetryHandler implements Interceptor{
+
+    private TelemetryHandlerOption _telemetryHandlerOption;
+
+    /**
+     * TelemetryHandler no param constructor
+     */
+    public TelemetryHandler() {
+        this(null);
+    }
+
+    /**
+     * TelemetryHandler constructor with passed in options.
+     * @param telemetryHandlerOption The user specified telemetryHandlerOptions
+     */
+    public TelemetryHandler(@Nullable TelemetryHandlerOption telemetryHandlerOption) {
+        if (telemetryHandlerOption == null) {
+            this._telemetryHandlerOption = new TelemetryHandlerOption();
+        }
+        this._telemetryHandlerOption = telemetryHandlerOption;
+    }
+
+    /**
+     * Send the HttpRequest after telemetryOptions are handled
+     * @param chain
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public Response intercept(@Nonnull Chain chain) throws IOException {
+        final Request request = chain.request();
+
+        TelemetryHandlerOption telemetryHandlerOption = request.tag(TelemetryHandlerOption.class);
+        if(telemetryHandlerOption == null) { telemetryHandlerOption = this._telemetryHandlerOption; }
+
+        //Simply forward request if TelemetryConfigurator is set to null intentionally.
+        if(telemetryHandlerOption == null) {
+            return chain.proceed(request);
+        }
+
+        //Use the TelemetryConfigurator set by the user to enrich the request as desired.
+        Request enrichedRequest = telemetryHandlerOption.TelemetryConfigurator.apply(request);
```

Comment:
today we don't have a way to add default options to the request adapter. And we don't want people to have to specify this for every call.
Here is what I suggest to align on other SDKs:
1. move this back to graph core
2. remove the configurator from the option
3. add this callback/lambda to the constructor as a parameter
4. assign from the constructor to a field
5. use that field on the request
6. provide an implementation of the lambda in graph core (you should already have one in facts)
7. add this middleware and pass it the lambda in the graph core default middleware.

Reply:
I believe we have a handler for when this is null meaning that it doesn't have to be specified for every call if it's not needed, this is essentially a mirror of what we are doing in the dotnet library. You are right though as we do need a telemetryHandler that is specific to graph like it's done here: 
https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/feature/3.0/src/Microsoft.Graph.Core/Requests/Middleware/GraphTelemetryHandler.cs
With this implementation the GraphTelemetryHandler is fully independent of the Kiota generic implementation of TelemetryHandler, should I mirror this approach in Java? That would mean leaving this implementation here as is now and doing a seperate implementation in Java-Core for v3.

Reply:
yep, please go ahead an mirror that in Java. I had forgotten we went this way in dotnet. let's align, it's going to be easier moving forward. Thanks for the callout.

Reply:
On it!